packages/devtools-local-toolbox/src/clients/chrome.js

```diff
@@ -38,51 +40,70 @@ function connectClient() {
   const deferred = defer();
 
   if(!getValue("chrome.debug")) {
-    return Promise.resolve(createTabs([]))
+    return Promise.resolve(createTabs([], {}))
   }
 
-  const webSocketPort = getValue("chrome.webSocketPort");
-  const url = `http://localhost:${webSocketPort}/json/list`;
-  networkRequest(url).then(res => {
-    deferred.resolve(createTabs(JSON.parse(res.content)))
-  }).catch(() => deferred.reject());
+  const port = getValue("chrome.port");
+  function fetch(http, options, callback) {
+    const {host, path, port} = options;
+    const url = `http://${host}:${port}${path}`;
+    return networkRequest(url).then(res => callback(null, res.content))
+  }
 
-  return deferred.promise;
+  return CDP.List({ port, fetch })
+    .then(tabs => createTabs(tabs, { browser: "chrome", type: "page" }))
+    .catch(err => {});
+}
+
+
+function connectNodeClient() {
+
+  if(true){
+    return Promise.resolve(createTabs([], {}))
+  }
+
+  const port = getValue("node.port");
+  return CDP.List({ port })
+    .then(tabs => createTabs(
+      JSON.parse(res.content),
+      { browser: "node", type: "node" }
+    ));
 }
 
 function connectTab(tab) {
-  return connect(tab.webSocketDebuggerUrl).then(conn => {
-    connection = conn;
+  const { webSocketDebuggerUrl } = tab;
+  debugger
+  return CDP({ webSocketDebuggerUrl }).then(_client => {
+    client = _client;
   });
 }
 
-function connectNode(url) {
-  return connect(url).then(conn => {
-    connection = conn;
-  });
+function connectNode(tab) {
+  return connectTab(tab);
 }
 
 function initPage(actions) {
-  const agents = connection._agents;
 
-  setupCommands({ agents: agents });
-  setupEvents({ actions, agents })
+  setupCommands({ client });
+  setupEvents({ actions, client })
 
-  agents.Debugger.enable();
-  agents.Debugger.setPauseOnExceptions("none");
-  agents.Debugger.setAsyncCallStackDepth(0);
+  client.Debugger.enable();
+  client.Debugger.setPauseOnExceptions("none");
+  client.Debugger.setAsyncCallStackDepth(0);
 
-  agents.Runtime.enable();
-  agents.Runtime.run();
+  client.Runtime.enable();
+  client.Runtime.run();
 
-  agents.Page.enable();
+  client.Page.enable();
 
-  connection.registerDispatcher("Debugger", clientEvents);
-  connection.registerDispatcher("Page", pageEvents);
+  debugger
+  // connection.registerDispatcher("Debugger", clientEvents);
+  // connection.registerDispatcher("Page", pageEvents);
```

Comment:
will need to do something like what we do [here](https://github.com/devtools-html/debugger.html/blob/master/packages/devtools-local-toolbox/src/clients/firefox.js#L94-L97) to listen to all the events.